### ISSUE TITLE
[spdlog] Adding cmake config files possible?

### ISSUE THREAD

**Issue by niosHD** [bincrafters/community#144]
### Description of Problem, Request, or Question
The current spdlog package does not contain the typical CMake config files (i.e., `spdlogConfig.cmake`, `spdlogConfigVersion.cmake`, `spdlogTargets.cmake`). Since the build system generates these files on install, is there any reason they are not included? Adding them would permit to use the regular `find_package(spdlog)` CMake syntax in consuming packages.

### Package Details (Include if Applicable)
* Package Name/Version: **spdlog/0.16.3**

**Comment by solvingj** [comment]
Generally speaking we're happy to try to include the necessary `findXXX.cmake` files for each package, although we typically have to write ourselves because I don't think I've seen one be generated by the build system.  I just re-ran the build for spdlog and I don't see where those are generated.  Can you give any additional details on where you see those files?

**Comment by niosHD** [comment]
These files are generated during the install step of the library by cmake. For example, if you perform the following steps you will see them in `_install/lib64/cmake/spdlog`:
```bash
# fetch the source in your preferred way (e.g., via git )
$ git clone https://github.com/gabime/spdlog.git
$ cd spdlog
# generate a build system with cmake to use the install target
$ cmake -DBUILD_TESTING=OFF -DCMAKE_INSTALL_PREFIX=$(pwd)/_install .
$ cmake --build . --target install
```
Interestingly, this also gives you a pkgconfig file. Note however, that this file additionally requires [some patching](http://docs.conan.io/en/latest/reference/tools.html?highlight=replace_prefix_in_pc_file#tools-replace-prefix-in-pc-file) in order to be useful.

In my own projects I usually always reuse the install target as [documented here](http://docs.conan.io/en/latest/howtos/cmake_install.html). Given that install is implemented properly you get the package method for free. Is there any reason why you populate the package folder by hand?

**Comment by sourcedelica** [comment]
That's a good idea.  I'm going to do that on recipes in the future.  I did it in a recent `AutotoolsBuildEnvironment` project (which could use a `install()` method) and it made the packaging process much easier.

**Comment by solvingj** [comment]
FYI, I've re-worked the recipe to use the Cmake install.  Interestingly, this opens a number of new questions that we need to discuss internally.  Should have answers and a rebuild tomorrow.

**Comment by solvingj** [comment]
@niosHD i've rebuilt 0.16.3/stable using the cmake installer.  Please give it a try.

**Comment by niosHD** [comment]
Well, I have good and bad news. The good news is that your new cmake installer based package works like a charm for me and `find_package` works as indended. On the other hand, the bad new is that you updated the [wrong branch](https://github.com/bincrafters/conan-spdlog/network). Instead of [0.16.3/stable](https://bintray.com/bincrafters/public-conan/spdlog%3Abincrafters#files/bincrafters%2Fspdlog%2F0.16.3%2Fstable%2Fexport) you updated  [0.14.0/stable](https://bintray.com/bincrafters/public-conan/spdlog%3Abincrafters#files/bincrafters%2Fspdlog%2F0.14.0%2Fstable%2Fexport).

Anyway, thank you very much for your effort!

**Comment by solvingj** [comment]
You were right. I've now merged to 0.16.3/stable. Please let me know if it works on that version.

**Comment by niosHD** [comment]
Works like a charm. Thank you!
